tools/booster/Proxy.hs

```diff
@@ -347,26 +376,22 @@ respondEither mbStatsVar forceFallback booster kore req = case req of
                     | KoreJson.KJBottom _ <- simplified.state.term ->
                         pure (Left simplified.logs)
                     | otherwise -> do
-                        -- to convert back to a term/constraints form,
-                        -- we run a trivial execute request (in booster)
-                        -- We cannot call the booster internaliser without access to the server state
-                        -- This call would fail for a \bottom state
-                        let request =
-                                (emptyExecuteRequest simplified.state)
-                                    { _module = mbModule
-                                    , maxDepth = Just $ Depth 0
-                                    }
-                        Log.logInfoNS "proxy" "Making 0-step execute request to convert back to a term/constraints form"
-                        result <- booster $ Execute request
-                        case result of
-                            Right (Execute ExecuteResult{state = finalState}) ->
-                                -- return state converted by Booster and logs from simplification.
-                                -- The 0-step execute result won't have logs (no logs are requested)
-                                pure $ Right (finalState, simplified.logs)
-                            other -> do
+                        -- convert back to a term/constraints form by re-internalising
+                        let internalPattern =
+                                runExcept $
+                                    internalisePattern
+                                        DisallowAlias
+                                        IgnoreSubsorts
+                                        Nothing
+                                        def
+                                        simplified.state.term
+                        case internalPattern of
+                            Right p ->
+                                pure $ Right (Booster.toExecState p, simplified.logs)
+                            Left err -> do
```

Comment:
This change flips the order of the typical `a <Int b ==Bool true` predicates (`true` comes first now, which is what booster prefers. Might rework that preference to align with the `kore-rpc` behaviour.